rust/kernel/error.rs

```diff
@@ -5,60 +5,77 @@
 //! C header: [`include/uapi/asm-generic/errno-base.h`](../../../include/uapi/asm-generic/errno-base.h)
 
 use crate::str::CStr;
-use crate::{bindings, c_types};
+use crate::{bindings, c_types, static_assert};
 use alloc::{alloc::AllocError, collections::TryReserveError};
 use core::convert::From;
 use core::fmt;
-use core::num::TryFromIntError;
+use core::num::{NonZeroI32, TryFromIntError};
 use core::str::{self, Utf8Error};
 
 /// Generic integer kernel error.
 ///
 /// The kernel defines a set of integer generic error codes based on C and
 /// POSIX ones. These codes may have a more specific meaning in some contexts.
 ///
+/// We use NonZeroI32 as the inner type because Result<(), Error> can fit in
+/// 32 bit memory, and it takes one register (rather than 2 for i16 or i32)
+/// for returning or argument passing which is fast in many architectures.
```

Comment:
```suggestion
/// We use [`NonZeroI32`] as the inner type because `Result<(), Error>` can fit in
/// 32-bit memory, and it takes one register (rather than 2 for [`i16`] or [`i32`])
/// for returning or argument passing which is fast in many architectures.
```